firefox-ios/Client/Frontend/Summary/LiteLLMClientError.swift

```diff
@@ -0,0 +1,29 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+/// Errors produced by LiteLLMClient, with user-friendly descriptions.
+public enum LLMClientError: LocalizedError {
+    case requestCreationFailed
+    case invalidResponse
+    case noContent
+    case decodingFailed
+    case networkError(underlying: Error)
+
+    public var errorDescription: String? {
```

Comment:
```suggestion
    public var errorDescription: String {
```
will this ever be nil if we cover all cases?

Reply:
Updated in https://github.com/mozilla-mobile/firefox-ios/pull/27779/commits/09bfb49e63018a1e72b8cfe4128433b13fc63031